fieldservice_route/models/fsm_order.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+from odoo import api, models
+
+
+class FSMOrder(models.Model):
+    _inherit = 'fsm.order'
+
+    @api.model
+    def create(self, vals):
+        res = super(FSMOrder, self).create(vals)
+        fsm_route_obj = self.env['fsm.route']
+        if res.person_id and res.scheduled_date_start:
+            fsm_route = fsm_route_obj.search([
+                ('person_id', '=', res.person_id.id),
+                ('scheduled_date', '=', res.scheduled_date_start)], limit=1)
+            if not fsm_route:
+                fsm_route_obj.create({
+                    'person_id': res.person_id.id,
+                    'scheduled_date': res.scheduled_date_start,
+                    'date': datetime.now()
+                })
+        for fsm_route_id in fsm_route_obj.search([('order_ids', 'in', res.id)]):
+            if fsm_route_id.days:
+                fsm_route_id.days = False
+        return res
```

Comment:
@nikul-serpentcs You need to extract this logic into his own method so you can call it from `create()` and `write()`

Reply:
Done

Reply:
@murtuzasaleh Add a new method `_manage_fsm_route()` to include this logic and then call this method from `create()` and `write()`